utils/probing.py

```diff
@@ -0,0 +1,206 @@
+from typing import Iterable, Tuple
+from sklearn.model_selection import train_test_split
+from sklearn.linear_model import LogisticRegression
+from sklearn.pipeline import Pipeline, make_pipeline
+from sklearn.preprocessing import StandardScaler
+import copy
+
+import os
+import torch
+import numpy as np
+import pickle
+
+from transformer_lens import HookedTransformer, ActivationCache
+from jaxtyping import Float, Int
+from torch import Tensor
+
+
+def get_layer_act(
+    model: HookedTransformer, prompts_tokens: Int[Tensor, "batch pos"], layer: int,
+    mb_size: int = 16
+) -> Float[np.ndarray, "batch pos d_model"]:
+    """
+    Gets resid_post for a given position and layer.
+    """
+    layer_act_list = []
+    for batch_start in range(0, prompts_tokens.shape[0], mb_size):
+        _, cache = model.run_with_cache(prompts_tokens[batch_start:batch_start+mb_size], return_type=None)
+        layer_resid = cache[f'blocks.{layer}.hook_resid_post']
+        layer_act_list.append(layer_resid.squeeze().cpu().numpy())
+
+    layer_act = np.concatenate(layer_act_list, axis=0)
+    return layer_act
+
+
+def cache_and_save_component_activations(
+    model: HookedTransformer, tokens: Int[Tensor, "batch pos"], component: str, pos: int, mb_size: int = 16
+):
+    """
+    Gets resid_post for a given position across all layers
+    """
+    batch_list = []
+    for i in range(0, tokens.shape[0], mb_size):
+        _, cache = model.run_with_cache(tokens[i:i+mb_size], return_type=None)
+        layer_list = []
+        for l in range(model.cfg.n_layers):
+            layer_act = cache[f'blocks.{l}.{component}'][:, pos, :]
+            layer_list.append(layer_act.unsqueeze(1).cpu().numpy())
+        layer_list: Float[np.ndarray, "batch layer dim"] = np.concatenate(layer_list, axis=1)
+        batch_list.append(layer_list)
+    batch_list: Float[np.ndarray, "batch layer dim"] = np.concatenate(batch_list, axis=0)
+    print(f"Activation shape: {batch_list.shape}")
+    # save as pkl file
+    with open(f"data/cached_activations/2_8b_mood_inference/{component}_pos_{pos}_activations.pkl", "wb") as f:
+        pickle.dump(batch_list, f)
+
+    return batch_list
+
+
+def train_probe_at_layer_pos(
+    model: HookedTransformer, act_folder: str, labels: Int[np.ndarray, "batch"], layer: int, pos: int = 0, 
```

Comment:
`model` appears unused? Do you want to add model to the filename to avoid different models reading/writing to the same file?